From 3f2382921d3e27d0e7bc764b8fff5aa264937439 Mon Sep 17 00:00:00 2001
From: JC Fillion-Robin <jcfr@nvidia.com>
Date: Tue, 10 Mar 2026 12:48:03 -0400
Subject: [PATCH 3/5] Use external dependencies - Modules and examples

Update cmake modules, examples, and source CMakeLists files to use
external library dependencies.

Patch-name: 0003-Fix-cmakelists.patch
Upstream-Status: Inappropriate [OE-specific]
Signed-off-by: Ilies CHERGUI <ichergui@nvidia.com>
Signed-off-by: JC Fillion-Robin <jcfr@nvidia.com>
---
 examples/CMakeLists.txt                       |  1 -
 .../distributed_pose_tree/cpp/CMakeLists.txt  |  6 ++-
 .../pose_tree_basic/cpp/CMakeLists.txt        |  6 ++-
 .../holoinfer/src/infer/onnx/CMakeLists.txt   |  6 ++-
 modules/holoviz/examples/demo/CMakeLists.txt  | 14 +-----
 .../holoviz/examples/depth_map/CMakeLists.txt | 14 +-----
 .../holoviz/tests/functional/CMakeLists.txt   | 14 +-----
 .../holoviz/thirdparty/imgui/CMakeLists.txt   | 44 +++++++------------
 src/CMakeLists.txt                            |  3 +-
 9 files changed, 33 insertions(+), 75 deletions(-)

diff --git a/examples/CMakeLists.txt b/examples/CMakeLists.txt
index 309724d..0ab3bcb 100644
--- a/examples/CMakeLists.txt
+++ b/examples/CMakeLists.txt
@@ -50,7 +50,6 @@ if(HOLOSCAN_CPP_EXAMPLES)
   add_subdirectory(gpu_resident_example)
   add_subdirectory(gpu_resident_multi_io)
   add_subdirectory(gpu_resident_input)
-  add_subdirectory(matx)
   add_subdirectory(ping_any)
   add_subdirectory(ping_conditional)
   add_subdirectory(ping_cycle)
diff --git a/examples/pose_tree/distributed_pose_tree/cpp/CMakeLists.txt b/examples/pose_tree/distributed_pose_tree/cpp/CMakeLists.txt
index 60d98e6..a552303 100644
--- a/examples/pose_tree/distributed_pose_tree/cpp/CMakeLists.txt
+++ b/examples/pose_tree/distributed_pose_tree/cpp/CMakeLists.txt
@@ -7,8 +7,10 @@ add_executable(distributed_pose_tree
 )
 target_link_libraries(distributed_pose_tree
   PUBLIC
-  holoscan::core
-  holoscan::pose_tree
+    holoscan::core
+    holoscan::pose_tree
+  PRIVATE
+    ucxx::ucxx
 )
 
 # Copy config file
diff --git a/examples/pose_tree/pose_tree_basic/cpp/CMakeLists.txt b/examples/pose_tree/pose_tree_basic/cpp/CMakeLists.txt
index f344275..85f2040 100644
--- a/examples/pose_tree/pose_tree_basic/cpp/CMakeLists.txt
+++ b/examples/pose_tree/pose_tree_basic/cpp/CMakeLists.txt
@@ -7,8 +7,10 @@ add_executable(pose_tree_basic
 )
 target_link_libraries(pose_tree_basic
   PUBLIC
-  holoscan::core
-  holoscan::pose_tree
+    holoscan::core
+    holoscan::pose_tree
+  PRIVATE
+    ucxx::ucxx
 )
 
 # Copy config file
diff --git a/modules/holoinfer/src/infer/onnx/CMakeLists.txt b/modules/holoinfer/src/infer/onnx/CMakeLists.txt
index e739ee8..ad75309 100644
--- a/modules/holoinfer/src/infer/onnx/CMakeLists.txt
+++ b/modules/holoinfer/src/infer/onnx/CMakeLists.txt
@@ -2,7 +2,9 @@
 # SPDX-FileCopyrightText: Copyright (c) 2024-2026 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
 # SPDX-License-Identifier: Apache-2.0
 
-find_package(ONNXRuntime REQUIRED)
+find_package(PkgConfig)
+pkg_check_modules(libonnxruntime REQUIRED IMPORTED_TARGET GLOBAL libonnxruntime>=1.21.0)
+find_package(dlpack 1.0 REQUIRED)
 add_library(holoinfer_ort SHARED core.cpp)
 add_library(holoscan::infer::onnx_runtime ALIAS holoinfer_ort)
 set_target_properties(holoinfer_ort PROPERTIES
@@ -14,7 +16,7 @@ set_target_properties(holoinfer_ort PROPERTIES
 )
 target_link_libraries(holoinfer_ort
     PRIVATE
-        ONNXRuntime::ONNXRuntime
+        PkgConfig::libonnxruntime
         CUDA::cudart
         dlpack::dlpack
         holoscan::logger
diff --git a/modules/holoviz/examples/demo/CMakeLists.txt b/modules/holoviz/examples/demo/CMakeLists.txt
index 32d3a41..8f48bba 100644
--- a/modules/holoviz/examples/demo/CMakeLists.txt
+++ b/modules/holoviz/examples/demo/CMakeLists.txt
@@ -3,18 +3,6 @@
 
 set(PROJECT_NAME holoviz_demo)
 
-# fetch the dependencies
-include(FetchContent)
-
-# stb
-FetchContent_Declare(
-    stb
-    GIT_REPOSITORY https://github.com/nothings/stb.git
-    GIT_TAG af1a5bc352164740c1cc1354942b1c6b72eacb8a
-    GIT_PROGRESS TRUE
-    )
-FetchContent_MakeAvailable(stb)
-
 add_executable(${PROJECT_NAME})
 
 target_sources(${PROJECT_NAME}
@@ -24,7 +12,7 @@ target_sources(${PROJECT_NAME}
 
 target_include_directories(${PROJECT_NAME}
     PRIVATE
-        ${stb_SOURCE_DIR}
+        ${stb_INCLUDE_DIRS}
     )
 
 target_link_libraries(${PROJECT_NAME}
diff --git a/modules/holoviz/examples/depth_map/CMakeLists.txt b/modules/holoviz/examples/depth_map/CMakeLists.txt
index e08c197..36f0314 100644
--- a/modules/holoviz/examples/depth_map/CMakeLists.txt
+++ b/modules/holoviz/examples/depth_map/CMakeLists.txt
@@ -3,18 +3,6 @@
 
 set(PROJECT_NAME holoviz_depth_map)
 
-# fetch the dependencies
-include(FetchContent)
-
-# stb
-FetchContent_Declare(
-    stb
-    GIT_REPOSITORY https://github.com/nothings/stb.git
-    GIT_TAG af1a5bc352164740c1cc1354942b1c6b72eacb8a
-    GIT_PROGRESS TRUE
-    )
-FetchContent_MakeAvailable(stb)
-
 add_executable(${PROJECT_NAME})
 
 target_sources(${PROJECT_NAME}
@@ -24,7 +12,7 @@ target_sources(${PROJECT_NAME}
 
 target_include_directories(${PROJECT_NAME}
     PRIVATE
-        ${stb_SOURCE_DIR}
+        ${stb_INCLUDE_DIRS}
     )
 
 target_link_libraries(${PROJECT_NAME}
diff --git a/modules/holoviz/tests/functional/CMakeLists.txt b/modules/holoviz/tests/functional/CMakeLists.txt
index 9b2fd41..1076f10 100644
--- a/modules/holoviz/tests/functional/CMakeLists.txt
+++ b/modules/holoviz/tests/functional/CMakeLists.txt
@@ -3,18 +3,6 @@
 
 set(PROJECT_NAME holoviz_functionaltests)
 
-# fetch the dependencies
-include(FetchContent)
-
-# stb
-FetchContent_Declare(
-    stb
-    GIT_REPOSITORY https://github.com/nothings/stb.git
-    GIT_TAG af1a5bc352164740c1cc1354942b1c6b72eacb8a
-    GIT_PROGRESS TRUE
-)
-FetchContent_MakeAvailable(stb)
-
 find_package(CUDAToolkit REQUIRED)
 find_package(X11 REQUIRED)
 find_package(Vulkan REQUIRED)
@@ -57,7 +45,7 @@ target_compile_definitions(${PROJECT_NAME}
 
 target_include_directories(${PROJECT_NAME}
     PRIVATE
-        ${stb_SOURCE_DIR}
+        ${stb_INCLUDE_DIRS}
     )
 
 target_link_libraries(${PROJECT_NAME}
diff --git a/modules/holoviz/thirdparty/imgui/CMakeLists.txt b/modules/holoviz/thirdparty/imgui/CMakeLists.txt
index 30b1bfc..b808399 100644
--- a/modules/holoviz/thirdparty/imgui/CMakeLists.txt
+++ b/modules/holoviz/thirdparty/imgui/CMakeLists.txt
@@ -6,29 +6,17 @@ include(GNUInstallDirs)
 
 find_package(Vulkan REQUIRED)
 
-set(imgui_TAG "f3373780668fba1f9bd64c208d05c20b781c9a39")
-FetchContent_Declare(
-  imgui
-  GIT_REPOSITORY https://github.com/ocornut/imgui.git
-  GIT_TAG "${imgui_TAG}"
-  GIT_PROGRESS TRUE
-  PATCH_COMMAND git apply "${CMAKE_CURRENT_SOURCE_DIR}/imgui.patch" || true
-  )
-
-message(STATUS "Fetching imgui@${imgui_TAG}")
-FetchContent_MakeAvailable(imgui)
-
 add_library(imgui_backend STATIC)
 
 target_sources(imgui_backend
   PRIVATE
-    ${imgui_SOURCE_DIR}/backends/imgui_impl_glfw.cpp
-    ${imgui_SOURCE_DIR}/backends/imgui_impl_vulkan.cpp
+    ${IMGUI_SOURCE_DIR}/backends/imgui_impl_glfw.cpp
+    ${IMGUI_SOURCE_DIR}/backends/imgui_impl_vulkan.cpp
   )
 
 target_include_directories(imgui_backend
   PUBLIC
-    ${imgui_SOURCE_DIR}
+    ${IMGUI_SOURCE_DIR}
   )
 
 target_compile_definitions(imgui_backend
@@ -50,16 +38,16 @@ add_library(holoscan::viz::imgui ALIAS holoviz_imgui)
 
 target_sources(holoviz_imgui
   PRIVATE
-    ${imgui_SOURCE_DIR}/imgui.cpp
-    ${imgui_SOURCE_DIR}/imgui_draw.cpp
-    ${imgui_SOURCE_DIR}/imgui_tables.cpp
-    ${imgui_SOURCE_DIR}/imgui_widgets.cpp
-    ${imgui_SOURCE_DIR}/misc/cpp/imgui_stdlib.cpp
+    ${IMGUI_SOURCE_DIR}/imgui.cpp
+    ${IMGUI_SOURCE_DIR}/imgui_draw.cpp
+    ${IMGUI_SOURCE_DIR}/imgui_tables.cpp
+    ${IMGUI_SOURCE_DIR}/imgui_widgets.cpp
+    ${IMGUI_SOURCE_DIR}/misc/cpp/imgui_stdlib.cpp
   )
 
 target_include_directories(holoviz_imgui
   PUBLIC
-    $<BUILD_INTERFACE:${imgui_SOURCE_DIR}>
+    $<BUILD_INTERFACE:${IMGUI_SOURCE_DIR}>
     $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>/holoviz/imgui
   )
 
@@ -70,19 +58,19 @@ set_target_properties(holoviz_imgui
 
 install(
   FILES
-    ${imgui_SOURCE_DIR}/imconfig.h
-    ${imgui_SOURCE_DIR}/imgui_internal.h
-    ${imgui_SOURCE_DIR}/imgui.h
-    ${imgui_SOURCE_DIR}/imstb_rectpack.h
-    ${imgui_SOURCE_DIR}/imstb_textedit.h
-    ${imgui_SOURCE_DIR}/imstb_truetype.h
+    ${IMGUI_SOURCE_DIR}/imconfig.h
+    ${IMGUI_SOURCE_DIR}/imgui_internal.h
+    ${IMGUI_SOURCE_DIR}/imgui.h
+    ${IMGUI_SOURCE_DIR}/imstb_rectpack.h
+    ${IMGUI_SOURCE_DIR}/imstb_textedit.h
+    ${IMGUI_SOURCE_DIR}/imstb_truetype.h
   DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/holoviz/imgui
   COMPONENT "holoscan-dependencies"
   )
 
 install(
   FILES
-    ${imgui_SOURCE_DIR}/misc/cpp/imgui_stdlib.h
+    ${IMGUI_SOURCE_DIR}/misc/cpp/imgui_stdlib.h
   DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/holoviz/imgui/misc/cpp
   COMPONENT "holoscan-dependencies"
   )
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index a2316cc..5c959e7 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -63,6 +63,7 @@ endfunction()
 # ##############################################################################
 # # Add library holoscan::spdlog_logger
 # ##############################################################################
+find_package(spdlog 1.14.1 REQUIRED)
 add_holoscan_library(spdlog_logger common/logger/spdlog_logger.cpp)
 target_link_libraries(spdlog_logger
     PUBLIC
@@ -387,7 +388,7 @@ target_link_libraries(core
         tl::expected
         yaml-cpp::yaml-cpp
     PRIVATE
-        hwloc
+        PkgConfig::hwloc
         gRPC::grpc++
         gRPC::grpc++_reflection
         protobuf::libprotobuf
